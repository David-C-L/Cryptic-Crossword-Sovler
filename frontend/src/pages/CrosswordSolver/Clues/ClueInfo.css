.clueInfo {
  overflow-y: scroll;
  max-height: 30vh;
  min-height: 20vh;
}