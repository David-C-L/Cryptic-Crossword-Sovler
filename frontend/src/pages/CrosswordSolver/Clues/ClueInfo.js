import './ClueInfo.css';

function ClueInfo(props) {

    return (
        <div className="clueInfo">
            <h4 className="mb-3 text-center">{props.clueNo}, {props.isAcross ? "Across" : "Down"}</h4>
            {props.hasAnswer
            ?
            <div className="container">
                <div className="row">
                    <div className="col-3">
                        <h6 className="mb-3 text-center">Answer:</h6>
                    </div>
                    <div className="col-9">
                        <p>{props.answer}</p>
                    </div>
                </div>
                <div className="row">
                    <div className="col-3">
                        <h6>Confidence:</h6>
                    </div>
                    <div className="col-9">
                        <p>{props.confidence}</p>
                    </div>
                </div>
                <div className="row">
                    <div className="col-3">
                        <h6>Explanation:</h6>
                    </div>
                    <div className="col-9">
                        <p>{props.explanation}</p>
                    </div>
                </div>
            </div>
            :
            <div className="alert alert-warning text-center">
                <strong>Oh no!</strong> No answer nor explanation has been found yet.
            </div>
            }
        </div>
    );

}

export default ClueInfo;