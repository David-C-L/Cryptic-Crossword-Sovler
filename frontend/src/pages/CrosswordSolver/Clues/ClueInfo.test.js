import React from "react";
import { unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";
import ClueInfo from "./ClueInfo";


let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Router><ClueInfo /></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});