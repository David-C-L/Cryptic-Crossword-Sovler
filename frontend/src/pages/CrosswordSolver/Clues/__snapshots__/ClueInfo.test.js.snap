// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="clueInfo"
>
  <h4
    className="mb-3 text-center"
  >
    , 
    Down
  </h4>
  <div
    className="alert alert-warning text-center"
  >
    <strong>
      Oh no!
    </strong>
     No answer nor explanation has been found yet.
  </div>
</div>
`;
